packages/next/src/build/babel/loader/get-config.ts

```diff
@@ -159,6 +159,95 @@ function getCustomBabelConfig(configFilePath: string) {
   )
 }
 
+let babelConfigWarned = false
+/**
+ * Check if custom babel configuration from user only contains options that
+ * can be migrated into latest Next.js features supported by SWC.
+ *
+ * This raises soft warning messages only, not making any errors yet.
+ */
+function checkCustomBabelConfigDeprecation(
+  config: Record<string, any> | undefined
+) {
+  if (!config || Object.keys(config).length === 0) {
+    return
+  }
+
+  const { plugins, presets, ...otherOptions } = config
+  if (Object.keys(otherOptions ?? {}).length > 0) {
```

Comment:
`otherOptions` is guaranteed to be an object.

```suggestion
  if (Object.keys(otherOptions).length > 0) {
```